pi_ldapproxy/proxy.py

```diff
@@ -218,19 +226,25 @@ def handleBeforeForwardRequest(self, request, controls, reply):
         :return:
         """
         if isinstance(request, pureldap.LDAPBindRequest):
-            if self.bound:
-                log.warn('Rejected a second bind request in the same connection')
-                self.send_bind_response((False, 'Reusing connections currently unsupported.'), request, reply)
-                return None
-            elif request.dn == '':
+            if self.received_bind_request:
+                # We have already received a bind request in this connection!
+                if self.factory.allow_connection_reuse:
+                    # We need to reset the state before further processing the request
+                    log.info('Reusing LDAP connection, resetting state ...')
+                    self.reset_state()
+                else:
+                    log.warn('Rejected a second bind request in the same connection')
+                    self.send_bind_response((False, 'Reusing connections currently unsupported.'), request, reply)
+                    return None
+            self.received_bind_request = True
```

Comment:
I think I do not get the logic here. Whey set received_bind to True in all cases?

Reply:
We only execute the ``if`` branch if we have received a bind request -- so we can set ``received_bind_request`` to True in any case then. We need this value to determine that the LDAP connection is reused if we receive another BIND request in the same connection later.

Reply:
Ah, I see the top level if-branch!